diff --git a/test/e2e/source_sinkbinding_v1_test.go b/test/e2e/source_sinkbinding_v1_test.go
index 7b3cbd311..6aa62c4de 100644
--- a/test/e2e/source_sinkbinding_v1_test.go
+++ b/test/e2e/source_sinkbinding_v1_test.go
@@ -42,6 +42,7 @@ import (
 )
 
 func TestSinkBindingV1CronJob(t *testing.T) {
+	t.Skip("SRVKE-500: Skipping since we set bindings to inclusion")
 	const (
 		sinkBindingName = "e2e-sink-binding"
 		deploymentName  = "e2e-sink-binding-cronjob"
