diff --git a/test/rekt/sink_binding_test.go b/test/rekt/sink_binding_test.go
index a77fb1403..536f54965 100644
--- a/test/rekt/sink_binding_test.go
+++ b/test/rekt/sink_binding_test.go
@@ -45,6 +45,7 @@ func TestSinkBindingV1Deployment(t *testing.T) {
 }
 
 func TestSinkBindingV1Job(t *testing.T) {
+        t.Skip("SRVKE-500: Skipping since we set bindings to inclusion")
 	t.Parallel()
 
 	ctx, env := global.Environment(
