diff --git a/test/rekt/broker_test.go b/test/rekt/broker_test.go
index 3b01c8dc2..06d824591 100644
--- a/test/rekt/broker_test.go
+++ b/test/rekt/broker_test.go
@@ -55,6 +55,7 @@ func TestBrokerAsMiddleware(t *testing.T) {
 
 // TestBrokerDLQ
 func TestBrokerWithDLQ(t *testing.T) {
+	t.Skip("skipping for test reasons...")
 	ctx, env := global.Environment(
 		knative.WithKnativeNamespace(system.Namespace()),
 		knative.WithLoggingConfig,
