diff --git a/vendor/knative.dev/reconciler-test/pkg/environment/images.go b/vendor/knative.dev/reconciler-test/pkg/environment/images.go
index 1cf5af4e5..8ad5c0b0c 100644
--- a/vendor/knative.dev/reconciler-test/pkg/environment/images.go
+++ b/vendor/knative.dev/reconciler-test/pkg/environment/images.go
@@ -90,7 +90,7 @@ func ProduceImages(ctx context.Context) (map[string]string, error) {
 		}
 		store.refs[koPack] = strings.TrimSpace(image)
 	}
-	return store.refs, nil
+	return store.copyRefs(), nil
 }
 
 func initializeImageStores(ctx context.Context) context.Context {
@@ -161,3 +161,11 @@ func (k imageStoreKey) get(ctx context.Context) *imageStore {
 type imageStore struct {
 	refs map[string]string
 }
+
+func (is *imageStore) copyRefs() map[string]string {
+	refs := make(map[string]string, len(is.refs))
+	for k, v := range is.refs {
+		refs[k] = v
+	}
+	return refs
+}
